eng/pipelines/templates/stages/platform-matrix.json

```diff
@@ -0,0 +1,36 @@
+{
+  "displayNames": {
+    "/p:UseProjectReferenceToAzureClients=true": "UseProjectRef",
+    "true": "Record"
+  },
+  "matrix": {
+    "Agent": {
+      "ubuntu-18.04": { "OSVmImage": "MMSUbuntu18.04", "Pool": "azsdk-pool-mms-ubuntu-1804-general" },
+      "windows-2019": { "OSVmImage": "MMS2019", "Pool": "azsdk-pool-mms-win-2019-general" }
+    },
+    "TestTargetFramework": [ "netcoreapp2.1", "net5.0" ]
+  },
+  "include": [
+    {
+      "Agent": {
+        "windows-2019": { "OSVmImage": "MMS2019", "Pool": "azsdk-pool-mms-windows-2019-general" }
+      },
+      "TestTargetFramework": "net461",
+      "AdditionalTestArguments": "/p:UseProjectReferenceToAzureClients=true"
+    },
+    {
+      "Agent": {
+        "windows-2019": { "OSVmImage": "MMS2019", "Pool": "azsdk-pool-mms-windows-2019-general" }
+      },
+      "TestTargetFramework": "netcoreapp2.1",
+      "SupportsRecording": true
+    },
+    {
+      "Agent": {
+        "macOS-10.15": { "OSVmImage": "macOS-10.15", "Pool": "Azure Pipelines" }
```

Comment:
We should put this in the root matrix as well as macOS does support net5.0 as well, we just didn't want to include it because of resourcing but the sparse matrix work should help with that.